### DISCUSSION TITLE
TTS -> SRS service?

### DISCUSSION THREAD

**Discussion by rkusa** [DCS-gRPC/rust-server#70] [Ideas]
Should we consider adding a service that receives text and transmits the text as speech over a specified frequency on SRS?

Some initial comments were made in:
- https://github.com/DCS-gRPC/rust-server/discussions/67#discussioncomment-1580444 and
- https://github.com/DCS-gRPC/rust-server/discussions/67#discussioncomment-1584373

**Comment by rkusa** [comment]
https://github.com/DCS-gRPC/rust-server/discussions/67#discussioncomment-1584373

> I think we can generalize it by saying we need a wrapper for SRS_ExternalAutio.exe
> 
> Ciribob provides a LUA interface https://github.com/ciribob/DCS-SimpleTextToSpeech
> 
> But LUA isn't the best at running binaries, has some wierd limitations and issues. like charachter limit and not being able to launch background process..
> 
> so adding a "TransmitRpc" function, which will emit all the args in json format to gRPC which will launch the binary, should cover most cases.

@uriba107 from a pure technical perspective, I am not too keen on having the gRPC server spawn up an exe, especially as we'd be able to easily achieve the same without having to spawn an exe. [`DATIS`](https://github.com/rkusa/DATIS) already has a whole TTS (including Win TTS, GCloud, AWS, ... TTS) to SRS stack written in Rust which we could easily pull over if we decide that this is a use-case worth supporting.

**Reply by uriba107** [reply]
I wasn't aware. Then yes, it will make much more sense. It will also provide overlord (or any other similar service) another TTS provider "for free".

**Reply by rurounijones** [reply]
I am not a fan of baking this into the server, it is expanding the scope an awful lot. I would rather have a client do it. We could write the APIs on the server such that the client that does the SRS transmissions can be called from the MSE or another client (Similar to discussion in #67 )
